torchvision/prototype/datasets/_builtin/sintel.py

```diff
@@ -0,0 +1,180 @@
+
+    T = TypeVar("T")
+
+    def pairwise(iterable: Iterable[T]) -> Iterable[Tuple[T, T]]:
+        a, b = tee(iterable)
+        next(b, None)
+        return zip(a, b)
+
+
+class InSceneGrouper(IterDataPipe[Tuple[Tuple[str, T], Tuple[str, T]]]):
+    def __init__(self, datapipe: IterDataPipe[Tuple[str, T]]) -> None:
+        self.datapipe = datapipe
+
+    def __iter__(self) -> Iterator[Tuple[Tuple[str, Any], Tuple[str, Any]]]:
+        for item1, item2 in pairwise(sorted(self.datapipe)):
+            if pathlib.Path(item1[0]).parent != pathlib.Path(item2[0]).parent:
+                continue
+
+            yield item1, item2
+
+
+class SINTEL(Dataset):
+
+    _FILE_NAME_PATTERN = re.compile(r"(frame|image)_(?P<idx>\d+)[.](flo|png)")
+    _PASS_NAME_BOTH = re.compile(r"clean|final")
+
+    def _make_info(self) -> DatasetInfo:
+        return DatasetInfo(
+            "sintel",
+            type=DatasetType.IMAGE,
+            homepage="http://sintel.is.tue.mpg.de/",
+            valid_options=dict(
+                split=("train", "test"),
+                pass_name=("clean", "final", "both"),
+            ),
+        )
+
+    def resources(self, config: DatasetConfig) -> List[OnlineResource]:
+        archive = HttpResource(
+            "http://files.is.tue.mpg.de/sintel/MPI-Sintel-complete.zip",
+            sha256="bdc80abbe6ae13f96f6aa02e04d98a251c017c025408066a00204cd2c7104c5f",
+        )
+        return [archive]
+
+    def _filter_split(self, data: Tuple[str, Any], *, split: str) -> bool:
+        path = pathlib.Path(data[0])
+        # The dataset contains has the folder "training", while allowed options for `split` are
+        # "train" and "test", we don't check for equality here ("train" != "training") and instead
+        # check if split is in the folder name
+        return split in path.parents[2].name
+
+    def _filter_images(self, data: Tuple[str, Any], *, pass_name: str) -> bool:
+        path = pathlib.Path(data[0])
+        regexp = self._PASS_NAME_BOTH if pass_name == "both" else re.compile(pass_name)
+        matched = bool(regexp.search(str(path.parent)))
+        return matched and path.suffix == ".png"
+
+    def _classify_archive(self, data: Tuple[str, Any], *, pass_name: str) -> Optional[int]:
+        path = pathlib.Path(data[0])
+        suffix = path.suffix
+        if suffix == ".flo":
+            return 0
+        elif suffix == ".png":
+            regexp = self._PASS_NAME_BOTH if pass_name == "both" else re.compile(pass_name)
+            matched = regexp.search(path.parents[1].name)
+            return 1 if matched else None
+        else:
+            return None
+
+    def _read_flo(self, file: io.IOBase) -> torch.Tensor:
+        magic = file.read(4)
+        if magic != b"PIEH":
+            raise ValueError("Magic number incorrect. Invalid .flo file")
+        w = int.from_bytes(file.read(4), "little")
+        h = int.from_bytes(file.read(4), "little")
+        data = file.read(2 * w * h * 4)
```

Comment:
these are expected to be encoded as little endian as well

Reply:
This is a TODO left from the current comments. Working on it now. Thanks @NicolasHug for pointing this out.

Reply:
I changed the `dtype` from `np.float32` to `<f4` for `data_arr`, but just wanted to quickly check, is this the correct fix? cc: @pmeier

Reply:
Yes `<f4` would be correct, it's a little endian float that fits in 4 bytes (float32).

But it might be preferable to always call the same function instead of mixing `frombuffer` and `from_bytes`.

Reply:
Right, that makes sense. Thanks, @NicolasHug! I've fixed this (preferred using `frombuffer` in both instances).